enhancements/machine-api/cluster-autoscaler-operator.md

```diff
@@ -0,0 +1,290 @@
+
+	// How long a node should be unneeded before it is eligible for scale down
+	UnneededTime *string `json:"unneededTime,omitempty"`
+}
+```
+
+##### MachineAutoscaler
+
+A new CRD will be introduced called `MachineAutoscaler`.
+The specification of the `MachineAutoscaler` CRD will allow users to configure a MachineSet to be autoscaled by adding
+an object reference pointing to the desired MachineSet.
+It will also allow the user to provide maximum and minimum boundaries for the number of Machines that each MachineSet
+should be allowed to contain. Note these boundaries are inclusive.
+
+```
+// MachineAutoscalerSpec defines the desired state of MachineAutoscaler
+type MachineAutoscalerSpec struct {
+	// MinReplicas constrains the minimal number of replicas of a scalable resource
+	MinReplicas int32 `json:"minReplicas"`
+
+	// MaxReplicas constrains the maximal number of replicas of a scalable resource
+	MaxReplicas int32 `json:"maxReplicas"`
+
+	// ScaleTargetRef holds reference to a scalable resource
+	ScaleTargetRef CrossVersionObjectReference `json:"scaleTargetRef"`
+}
+
+// CrossVersionObjectReference identifies another object by name, API version,
+// and kind.
+type CrossVersionObjectReference struct {
+	// APIVersion defines the versioned schema of this representation of an
+	// object. Servers should convert recognized schemas to the latest internal
+	// value, and may reject unrecognized values. More info:
+	// http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#resources
+	APIVersion string `json:"apiVersion,omitempty"`
+
+	// Kind is a string value representing the REST resource this object
+	// represents. Servers may infer this from the endpoint the client submits
+	// requests to. Cannot be updated. In CamelCase. More info:
+	// http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#types-kinds
+	Kind string `json:"kind"`
+
+	// Name specifies a name of an object, e.g. worker-us-east-1a.
+	// Scalable resources are expected to exist under a single namespace.
+	Name string `json:"name"`
+}
+```
+
+The `MinReplicas` and `MaxReplicas` values will be copied onto the target `MachineSet` as annotations under
+`"machine.openshift.io/cluster-api-autoscaler-node-group-min-size"` and
+`"machine.openshift.io/cluster-api-autoscaler-node-group-max-size"` respectively.
+
+### Risks and Mitigations
+
+## Design Details
+
+### Test Plan
+
+### Graduation Criteria
+
+#### Examples
+
+##### Dev Preview -> Tech Preview
+
+##### Tech Preview -> GA
+
+##### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+## Implementation History
+
+## Drawbacks
+
+## Alternatives
+
+### Don't provide an operator
+
+In this case, users would need to manually configure a Deployment for the Cluster Autoscaler and will need to manually
+mark MachineSets with the required annotations for the Cluster Autoscaler to be able to identify which MachineSets are
+allowed to be autoscaled.
```

Comment:
is it worth mentioning the complexity of deploying and operating a cluster autoscaler here? seems like a win for the user to not need to understand all those options.